rust/main/agents/relayer/src/relayer.rs

```diff
@@ -620,7 +621,277 @@ impl Relayer {
+                        )
+                        .unwrap()
+                        .as_slice(),
+                    ),
+                    merkle_tree_hook: H256::from_slice(
+                        hex::decode(
+                            "000000000000000000000000AD34A66Bf6dB18E858F6B686557075568c6E031C",
+                        )
+                        .unwrap()
+                        .as_slice(),
+                    ),
+                },
+                connection: ChainConnectionConf::Ethereum(h_eth::ConnectionConf {
+                    rpc_connection: h_eth::RpcConnectionConf::Http {
+                        url: Url::parse("https://sepolia-rollup.arbitrum.io/rpc").unwrap(),
+                    },
+                    transaction_overrides: h_eth::TransactionOverrides {
+                        gas_price: None,
+                        gas_limit: None,
+                        max_fee_per_gas: None,
+                        max_priority_fee_per_gas: None,
+                    },
+                    operation_batch: OperationBatchConfig {
+                        batch_contract_address: None,
+                        max_batch_size: 1,
+                    },
+                }),
+                metrics_conf: PrometheusMiddlewareConf {
+                    contracts: HashMap::new(),
+                    chain: None,
+                },
+                index: IndexSettings {
+                    from: 0,
+                    chunk_size: 1,
+                    mode: IndexMode::Block,
+                },
+            },
+        )];
+
+        RelayerSettings {
+            base: Settings {
+                chains: chains.into_iter().collect(),
+                metrics_port: 5000,
+                tracing: TracingConfig::default(),
+            },
+            db: PathBuf::new(),
+            origin_chains: [
+                HyperlaneDomain::Known(KnownHyperlaneDomain::Arbitrum),
+                HyperlaneDomain::Known(KnownHyperlaneDomain::Ethereum),
+                HyperlaneDomain::Known(KnownHyperlaneDomain::Optimism),
+            ]
+            .into_iter()
+            .collect(),
+            destination_chains: [
+                HyperlaneDomain::Known(KnownHyperlaneDomain::Arbitrum),
+                HyperlaneDomain::Known(KnownHyperlaneDomain::Ethereum),
+                HyperlaneDomain::Known(KnownHyperlaneDomain::Optimism),
+            ]
+            .into_iter()
+            .collect(),
+            gas_payment_enforcement: Vec::new(),
+            whitelist: MatchingList::default(),
+            blacklist: MatchingList::default(),
+            address_blacklist: Vec::new(),
+            transaction_gas_limit: None,
+            skip_transaction_gas_limit_for: HashSet::new(),
+            allow_local_checkpoint_syncers: true,
+            metric_app_contexts: Vec::new(),
+        }
+    }
+
+    #[tokio::test]
+    #[tracing_test::traced_test]
+    async fn test_failed_build_mailboxes() {
```

Comment:
Turns out we don't need to do too many fancy things to cause an error for `build_mailboxes` or `build_validator_announces`.
All we need to do is just not have a `ChainConf` associated with said chain and it will cause an error in `build_mailbox / build_validator_announce`